source/MapDetailPanel.cpp

```diff
@@ -817,23 +817,44 @@ void MapDetailPanel::DrawInfo()
 	// Adapt the coordinates for the text (the sprite is drawn from a center coordinate).
 	uiPoint.X() -= (tradeSprite->Width() / 2. - textMargin);
 	uiPoint.Y() -= (tradeSprite->Height() / 2. - textMargin);
+
+	// Don't "compare" prices if the current system is uninhabited and thus has no prices to compare to.
+	bool noCompare = (!player.GetSystem() || !player.GetSystem()->IsInhabited(player.Flagship()));
+	int value = 0;
+	double lowCompare = 0;
+	double highCompare = 0;
+
+	// When comparing prices, determine min/max deltas in order to represent commodity delta prices for displayed
+	// commodities as a gradient.
+	if(!noCompare && canView && selectedSystem->IsInhabited(player.Flagship()))
+	{
+		for(const Trade::Commodity &commodity : GameData::Commodities())
+		{
+			value = selectedSystem->Trade(commodity.name);
+			int localValue = (player.GetSystem() ? player.GetSystem()->Trade(commodity.name) : 0);
+			if(value && localValue)
+			{
+				value -= localValue;
+				lowCompare = value < lowCompare ? value : lowCompare;
+				highCompare = value > highCompare ? value : highCompare;
```

Comment:
Maybe this will be clearer?
```suggestion
				if(value < lowCompare)
					lowCompare = value;
				if(value > highCompare)
					highCompare = value;
```